src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs

```diff
@@ -213,5 +239,13 @@ private static unsafe void GetCryptoRandomBytes(byte* bytes, int byteCount)
                 throw new InvalidOperationException(SR.InvalidOperation_Cryptography);
             }
         }
+
+        // This P/Invoke is normally handled by the crypto shim, but currently calling
+        // the Apple-based crypto shim requires cross-compiling, and that is a very big
+        // hammer for calling a very stable system API. If this assembly ever gets split
+        // such that OSX and Unix are different compilations, this should move to standard
+        // interop patterns.
+        [DllImport("/usr/lib/libSystem.B.dylib")]
+        private static unsafe extern int CCRandomGenerateBytes(byte* bytes, int byteCount);
```

Comment:
Could we still define this in the Common Interop folder, just so all of our production P/Invokes are defined there?  We already have several library defines directly to the macOS libs: https://github.com/dotnet/corefx/blob/ba10ab731e96641c2c6b418fa174097b50520605/src/Common/src/Interop/OSX/Interop.Libraries.cs#L7
This would be similar.